fabtests/pytest/common.py

```diff
@@ -33,17 +34,31 @@ def has_ssh_connection_err_msg(output):
 
     return False
 
-
+@functools.lru_cache(10)
```

Comment:
I think you should increase the lru_cache size to max number of hosts we would ever expect, or just leave it unspecified `@functools.lru_cache`, which defaults to 128.